ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -80,14 +83,44 @@ export const NetworkListMenu = ({ onClose }) => {
 
   const showSearch = nonTestNetworks.length > 3;
 
+  const orderedNetworksList = useSelector(getOrderedNetworksList);
+
+  const newOrderNetworks = () => {
+    if (orderedNetworksList.length === 0) {
+      return nonTestNetworks;
+    }
+
+    // Reorder nonTestNetworks based on the order of chainIds in orderedNetworksList
+    const sortedNetworkList = orderedNetworksList
+      .map((chainId) =>
+        nonTestNetworks.find((network) => network.chainId === chainId),
+      )
+      .filter((network) => network !== undefined);
```

Comment:
Could this be `filter(Boolean)` instead?

Reply:
Updated https://github.com/MetaMask/metamask-extension/pull/21163/commits/cd4049f31e56fa9cf0360157f96511629070e83e